docs/TCP_Network.txt

```diff
@@ -1,100 +1,99 @@
+3. She uses the onion through the TCP relay connections to send friend requests
+or tell online friends which TCP nodes she is connected to and her temporary
 public key.
 
-4. Bob receives an onion packet from Alice telling him which nodes she is 
-connected to. Bob connects to these nodes and establishes a routed connection 
+4. Bob receives an onion packet from Alice telling him which nodes she is
+connected to. Bob connects to these nodes and establishes a routed connection
 with alice using that temporary public key.
 
-5. That connection is used by both to transmit encrypted Messenger and A/V 
+5. That connection is used by both to transmit encrypted Messenger and A/V
 packets.
 
-6. If one of the nodes shuts down while it is currently routing traffic, Alice 
+6. If one of the nodes shuts down while it is currently routing traffic, Alice
 and bob just switch to one of the other nodes they are both connected to.
 
 
 Detailed implementation details:
 
-There are two distinct parts for TCP relays, the client part and the server 
+There are two distinct parts for TCP relays, the client part and the server
 part.
 
-The server acts as the actual relay. Servers must have fully forwarded TCP 
-ports (NAT-PMP and uPNP can help here). The first port the server will try 
-binding to is 443 followed by port 3389 and possibly some others. Onion packets 
+The server acts as the actual relay. Servers must have fully forwarded TCP
+ports (NAT-PMP and uPNP can help here). The first port the server will try
+binding to is 443 followed by port 3389 and possibly some others. Onion packets
 can be sent/received through the TCP servers.
 
 
 Server:
 
-The public/private key pair the TCP server uses is the same one he uses for the 
-DHT. 
+The public/private key pair the TCP server uses is the same one he uses for the
+DHT.
 
-all crypto for communication with the server uses the crypto_box() function of 
+All crypto for communication with the server uses the crypto_box() function of
 NaCl.
 
-TCP doesn't have packets so what we will refer to as packets are sent this way: 
+TCP doesn't have packets so what we will refer to as packets are sent this way:
 [[uint16_t (length of data)][data]]
 
 So if you would inspect the TCP stream you would see:
-[[uint16_t (length of data)][data]][[uint16_t (length of 
+[[uint16_t (length of data)][data]][[uint16_t (length of
 data)][data]][[uint16_t (length of data)][data]]
 
-Note that both handshake packets don't have this format (the length for them is 
+Note that both handshake packets don't have this format (the length for them is
 always the same so we don't need to specify it.)
 
 When the client connects to the server, he sends this packet:
-[public key of client (32 bytes)][nonce for the encrypted data [24 
-bytes]][encrypted with the private key of the client and public key of the 
-server and the nonce:[public key (32 bytes) and][base nonce we want the server 
-to use to encrypt the packets sent to us (24 bytes)]]
```

Comment:
"base nonce we want the server to use to encrypt the packets sent to us (24 bytes)"
This actually seems to be the nonce, the server uses to *de*crypt the packets. Is this intended?
Same goes for the handshake response from the server.